src/mpid/ch4/netmod/ofi/init_settings.c

```diff
@@ -164,6 +164,12 @@ int MPIDI_OFI_init_hints(struct fi_info *hints)
         }
 #endif
 
+#ifdef FI_MR_HMEM
+        if (MPIDI_OFI_ENABLE_HMEM && MPIDI_OFI_ENABLE_MR_HMEM) {
+            hints->domain_attr->mr_mode |= FI_MR_HMEM;
+        }
+#endif
+
```

Comment:
AR: Need to follow up with libfabric folks to see if this is needed as input or we can always query MR_HMEM

Reply:
Sorry for the delay on this. I've made some changes based on our previous discussions:

* Always check `FI_MR_HMEM` result from provider if `FI_HMEM` is enabled
* Allow the user to force buffer registration via `MPIR_CVAR_CH4_OFI_ENABLE_MR_HMEM` (value of `-1` == use provider result)